src/Serialization/Address.purs

```diff
@@ -1,376 +1,283 @@
-module Serialization.Address
-  ( class Address
-  , AddressCsl
-  , BaseAddressCsl
-  , BaseAddress
-  , RewardAddress
-  , RewardAddressCsl
-  , PointerAddressCsl
-  , NetworkTag(TestnetTag, MainnetTag)
-  , StakeCred
-  , CredType
-  , StakeCredentialCsl
-  , PubKeyAddress
-  , ScriptAddress
-  , addressBech32
-  , addressBytes
-  , addrR
-  , baseAddressFromBech32
-  , baseAddressFromBytes
-  , rewardAddressFromBytes
-  , rewardAddressFromBech32
-  , ed25519KeyHashCredType
-  , baseAddress
-  , rewardAddress
-  , netTagToInt
-  , pubKeyAddress
-  , scriptAddress
-  , scriptHashCredType
-  , networkTag
-  , networkTagInt
-  , paymentCred
-  , delegationCred
-  , toAddressCsl
-  , intToNetTag
-  ) where
+module Serialization.Address where
 
 import Prelude
 
+import Control.Alt ((<|>))
+import Data.Function (on)
 import Data.Maybe (Maybe(..))
+import Data.Newtype (class Newtype)
+import Data.Typelevel.Undefined (undefined)
+import Data.UInt (UInt)
 import FfiHelpers (MaybeFfiHelper, maybeFfiHelper)
-import Serialization.Csl (class ToCsl, CslType, toCslRep, toCslType)
 import Serialization.Hash (Ed25519KeyHash, ScriptHash)
-import Types.Aliases (Bech32String)
+import Types.Aliases (Bech32String, Base58String)
 import Types.ByteArray (ByteArray)
 
--- | Denotes one of two possible values of Shelley address' network tag.
-data NetworkTag = TestnetTag | MainnetTag
+newtype Slot = Slot UInt
```

Comment:
Why change from `BigInt` to `UInt`? The Plutus `Slot` is a wrapped `Integer`

Reply:
It's based on cardano serialization library which uses UInt32. I think it is safer to respect this 'lower bound'

Reply:
Created an issue for this.

Reply:
OK, I think we should keep your changes for now then. If the Plutus representation and CSL diverge, we should favor the latter, I think. But we can discuss that more and change this later if we decide otherwise

Reply:
Addressed in an issue